main/main.c

```diff
@@ -1796,6 +1796,10 @@ void php_request_shutdown(void *dummy)
 
 	EG(flags) |= EG_FLAGS_IN_SHUTDOWN;
 
+	if (EG(stack_overflow)) {
+		zend_handle_stack_overflow();
+	}
```

Comment:
We should probably kill the thread or process after that, because it may be corrupted